compiler/damlc/daml-opts/daml-opts/DA/Daml/Options.hs

```diff
@@ -287,6 +287,7 @@ wOptsSet =
 --  , Opt_WarnPrepositiveQualifiedModule
   , Opt_WarnOverlappingPatterns
   , Opt_WarnIncompletePatterns
+  , Opt_WarnUnusedMatches
```

Comment:
I’m not sure we want to turn it on by default. We don’t even turn on other unused warnings (e.g. unused top-level binds). I would suggest that we split things in two: In this PR make the warning usable and let people enable it manually. Then in the future we can decide to enable this and other unused warnings by default.

Reply:
✔️